File: work-platform-react/src/util/DateUtil.js
import dayjs from "dayjs";

const getWeekRange = (weekNumber,year)=>{
    if (!weekNumber || !year) {
        return {};
    }
    const date = dayjs().year(year).week(weekNumber).day(1);
    const endDate = date.date(date.date() + 6);
    return{
        startDate : date.format('YYYY-MM-DD'),
        endDate : endDate.format('YYYY-MM-DD')
    }
}

const getWeekRangeByDate = (date)=>{
    if (!date){
        return '';
    }
    const startDate = dayjs(date);
    const endDate = startDate.add(6,'day');
    return{
        startDate : startDate.format('YYYY-MM-DD'),
        endDate : endDate.format('YYYY-MM-DD')
    }
}

const DateUtil = {
    getWeekRange,
    getWeekRangeByDate
}

export default DateUtil

File: work-platform-server/src/main/java/com/personalwork/util/DateUtil.java
package com.personalwork.util;

import java.time.DayOfWeek;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.time.temporal.TemporalAdjusters;

/**
 * @author 姚礼林
 * @desc TODO
 * @date 2024/7/10
 */
public class DateUtil {
    private static final DateTimeFormatter  FORMATTER =  DateTimeFormatter.ofPattern("yyyy-MM-dd");
    public static String[] getDateRangeByWeekBaseMonday(int weekNumber) {
        LocalDate nowDate = LocalDate.now();
        LocalDate endOfWeek = nowDate.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY));
        LocalDate startOfWeek = endOfWeek.minusWeeks(weekNumber);
        return new String[]{startOfWeek.format(FORMATTER), endOfWeek.format(FORMATTER)};
    }

    public static String[] getDateRangeByMonth(int monthNumber){
        LocalDate nowDate = LocalDate.now();
        LocalDate startDate = nowDate.minusMonths(monthNumber).withDayOfMonth(1);
        return new String[]{startDate.format(FORMATTER), nowDate.format(FORMATTER)};
    }


    public static String getNowDateStr() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
        LocalDate dateObj = LocalDate.now();
        return dateObj.format(formatter);
    }
}
